### ISSUE TITLE
Add support for pactos braille display drivers in to nvda core.

### ISSUE THREAD

**Issue by alireza1385mamani** [nvaccess/nvda#16361]
### Is your feature request related to a problem? Please describe.
Pactos, witch is an Iranian company that makes braille displays, made an nvda addon that could be used as a driver for all of its' braille displays, but because of the breaking point of the nvda 2024.1  this addon can't be used for these displays.
### Describe the solution you'd like
I'd like to see if we could implement the functions of this driver in to the core of the nvda.
### Describe alternatives you've considered
If we can not do that, I've created a repository [heare](https://github.com/alireza1385mamani/pactos) that we can work on and port it to python 3.11.
### Additional context
Hear is the addon file.
[pactos-1.0.5_beta.zip](https://github.com/nvaccess/nvda/files/14862408/pactos-1.0.5_beta.zip)

**Comment by gerald-hartig** [comment]
@alireza1385mamani can you please confirm the permissions on the link https://github.com/alireza1385mamani/pactos. It isn't accessible. 

@alireza1385mamani have you confirmed that the add-on no longer works in 2024.1? For any add-on in the Installed Add-Ons tab of the Add-On store noted as "Disabled (incompatible)" you have the option in the actions menu for that add-on to "Enable (override incompatibility)." 

Is it confirmed that Pactos will no longer update their add-on? 

Generally speaking we don't include features in core simply because an add-on developer has stopped maintaining their code. There are several factors that go into that decision. In this case the developer of the add-on should first be encouraged to update their code.

**Comment by alireza1385mamani** [comment]
[Hear](https://github.com/alireza1385mamani/pactos) is the public ripository. I've contacted pactos company several times and they didn't answer me at all. That's why I've asumed they don't want to work on it anymore.

**Comment by Adriani90** [comment]
I think generally braille display drivers were always a good candidate for the NVDA core.  It would ofcourse make totally sense if the vendors themselves contribute directly.   there is a wiki article  on how to contribute at integrating drivers for braille displays into the core. Maybe you can send this to the vendor as well.
 
 
https://github.com/nvaccess/nvda/wiki/AddingSupportForABrailleDisplay

if the company is not responsive, it is indeed tnot a good advertisment for their maintenance strategy.

**Comment by alireza1385mamani** [comment]
@Adriani90 You are correct about the fact that a companies responsiveness can effect there reputation, but whenever their is a monopoly in the industry, it becomes broken. I have to contact them on saturday to see what their answer is.

**Comment by gerald-hartig** [comment]
@alireza1385mamani Thanks for updating the link to the respository and contacting the developers. Have you encountered any issues with the driver and confirmed that the add-on no longer works in 2024.1? Whenever we do a 20xx.1 release, many add-ons are listed as incompatible but actually work fine.

**Comment by alireza1385mamani** [comment]
> @alireza1385mamani Thanks for updating the link to the respository and contacting the developers. Have you encountered any issues with the driver and confirmed that the add-on no longer works in 2024.1? Whenever we do a 20xx.1 release, many add-ons are listed as incompatible but actually work fine.

it is showing as beeing compatible and enabled in the addon store, but it isn't working at all.

**Comment by gerald-hartig** [comment]
The required .dll without source code is going to be a blocker to getting this into core. Given that the add-on source code is available, the best way forward at this point would be to get the add-on working first. 

Also, if the add-on is shown as being compatible, who updated it's manifest to 2024.1?

**Comment by alireza1385mamani** [comment]
They did it when they tried to release the 1.0.5

**Comment by seanbudd** [comment]
@michaelDCurran - we have some concerns about accepting a close sourced DLL into core, which is possibly a security risk or has incompatible license. any thoughts? otherwise we may need to close this and encourage the community to take on add-on maintenance

**Comment by seanbudd** [comment]
Closing - without licensing approval from pactos, we cannot accept this.
